src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT DISTINCT cars.id AS car_id, cars.model AS model, "
+                + "manufacturer_id, "
+                + "manufacturers.name AS manufacturer_name, "
+                + "manufacturers.country AS country, "
+                + "drivers.id AS driver_id, drivers.name AS driver_name, "
+                + "drivers.license_number AS license_number "
+                + "FROM cars INNER JOIN cars_drivers "
+                + "ON cars.id = cars_drivers.car_id "
+                + "INNER JOIN manufacturers "
+                + "ON manufacturer_id = cars.manufacturer_id "
+                + "INNER JOIN drivers "
+                + "ON drivers.id = cars_drivers.driver_id "
+                + "WHERE cars_drivers.driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(setCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars by driver "
+                    + driverId, e);
+        }
+        for (int i = 0; i < cars.size(); i++) {
+            cars.get(i).setDrivers(getAllByCar(cars.get(i).getId()));
+        }
+        return cars;
+    }
+
+    private List<Driver> getAllByCar(Long carId) {
+        List<Driver> drivers = new ArrayList<>();
+        String query = "SELECT * FROM drivers "
+                + "INNER JOIN cars_drivers "
+                + "ON drivers.id = cars_drivers.driver_id "
+                + "WHERE cars_drivers.car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Driver driver = new Driver();
+                driver.setId(resultSet.getObject("id", Long.class));
+                driver.setName(resultSet.getString("name"));
+                driver.setName(resultSet.getString("license_number"));
+                drivers.add(driver);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all drivers by car "
+                    + carId, e);
+        }
+        return drivers;
+    }
+
+    private Car setCar(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Car getCar(ResultSet resultSet) throws SQLException {
```

цей метод нічого не сетить, він дістає машину з ResultSet

Reply:
Done